sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShuffleBufferCatalog.scala

```diff
@@ -177,7 +177,7 @@ class ShuffleBufferCatalog(
         // NOTE: Not synchronizing array buffer because this shuffle should be inactive.
         bufferIds.foreach { id =>
           tableMap.remove(id.tableId)
-          bufferIdToHandle.get(id).close()
+          bufferIdToHandle.remove(id).close()
```

Comment:
nit: Do we want to check for null when removing this before closing it?

Reply:
Yeah, that's probably a good idea. Let me push something

Reply:
Addressed here: https://github.com/NVIDIA/spark-rapids/pull/8070/commits/7a272ecc0ed79b9cd6b59449435c7693342693bb